packages/flutter/lib/src/widgets/widget_state.dart

```diff
@@ -8,6 +8,89 @@ import 'package:flutter/services.dart';
 
 // Examples can assume:
 // late BuildContext context;
+// late Set<WidgetState> states;
+
+/// This class allows [WidgetState] enum values to be combined
+/// using [WidgetStateOperators].
+///
+/// A [Map] with [WidgetStateMapKey] objects as keys can be used
+/// in the [WidgetStateProperty.fromMap] constructor to resolve to
+/// one of its values, based on the first key that [isSatisfiedBy]
+/// the current set of states.
+///
+/// {@macro flutter.widgets.WidgetStateProperty.WidgetStateMap}
```

Comment:
As far as the order of examples in this doc macro, my thoughts are that you don't _need_ to understand the operators in order to make a `WidgetStateMap`, but you definitely would need to understand the basics of a `WidgetStateMap` in order to know how to use the operators.